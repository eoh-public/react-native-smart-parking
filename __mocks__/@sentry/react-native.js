export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
