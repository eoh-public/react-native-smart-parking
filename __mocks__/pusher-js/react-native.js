const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
