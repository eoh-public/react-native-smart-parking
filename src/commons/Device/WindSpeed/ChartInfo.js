import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import Text from '../../Text';
import { CircleView } from '../../index';

const ChartInfo = memo(
  ({ color, title, isLastItem, index, onPress, boldText }) => {
    const marginStyle = {
      marginRight: isLastItem ? 0 : 24,
    };
    const onShowOne = useCallback(() => {
      onPress && onPress(index);
    }, [onPress, index]);

    return (
      <TouchableOpacity
        style={[styles.chartBottom, marginStyle]}
        onPress={onShowOne}
      >
        <CircleView size={8} backgroundColor={color} style={styles.dotChart} />
        <Text size={14} bold={boldText}>
          {title}
        </Text>
      </TouchableOpacity>
    );
  }
);

export default ChartInfo;

const styles = StyleSheet.create({
  chartBottom: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 16,
    paddingTop: 4,
    flexDirection: 'row',
    marginRight: 24,
  },
  dotChart: {
    marginRight: 8,
  },
});
