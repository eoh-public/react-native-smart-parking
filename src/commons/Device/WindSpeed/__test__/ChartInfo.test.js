import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { TouchableOpacity } from 'react-native';
import ChartInfo from '../ChartInfo';

describe('Test ChartInfo', () => {
  let wrapper;
  test('onItemClick MenuActionAddNew', () => {
    const onPressMock = jest.fn();
    act(() => {
      wrapper = renderer.create(<ChartInfo onPress={onPressMock} />);
    });
    const isntance = wrapper.root;
    const button = isntance.findByType(TouchableOpacity);
    renderer.act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
    expect(onPressMock.mock.calls.length).toBe(1);
  });
});
