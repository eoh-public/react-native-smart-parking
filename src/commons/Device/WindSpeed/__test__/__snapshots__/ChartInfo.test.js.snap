// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChartInfo onItemClick MenuActionAddNew 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginRight": 24,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingTop": 4,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 0,
          "height": 8,
          "width": 8,
        },
        undefined,
        Object {
          "marginRight": 8,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 14,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  />
</View>
`;
